File: QuickSeries/app/src/main/java/com/test/quickseries/resource/list/adapter/ResourceViewHolder.java
package com.test.quickseries.resource.list.adapter;

import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.test.quickseries.databinding.ResourceItemBinding;

/**
 * Created by elton on 2017-10-15.
 */

public class ResourceViewHolder extends RecyclerView.ViewHolder {

    ResourceItemBinding binding;

    public ResourceViewHolder(View itemView) {
        super(itemView);
        binding = DataBindingUtil.bind(itemView);
    }
}

File: MobileTest/app/src/main/java/com/quickseries/mobiletest/adapters/resource/ResourceViewHolder.kt
package com.quickseries.mobiletest.adapters.resource

import android.support.v7.widget.AppCompatImageView
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import com.facebook.drawee.view.SimpleDraweeView
import com.quickseries.mobiletest.R
import com.quickseries.mobiletestcore.models.ResourceModelBase
import com.squareup.picasso.Picasso

class ResourceViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    var itemClick: ResourceClickDelegate? = null
    private lateinit var resource: ResourceModelBase
    private var resourceImageView = itemView.findViewById<AppCompatImageView>(R.id.resourceImageView)
    private var titleTextView = itemView.findViewById<TextView>(R.id.titleTextView)

    init {
        itemView.isClickable = true
        itemView.setOnClickListener {
            itemClick?.invoke(resource)
        }
    }

    fun dataBind(resource: ResourceModelBase) {
        this.resource = resource
        titleTextView.text = resource.title
        Picasso.get()
            .load(resource.photo)
            .fit()
            .placeholder(R.drawable.progress_animation)
            .error(R.drawable.ic_no_image)
            .into(resourceImageView)
    }
}